application/shared-kernel/ApiCore/Endpoints/TrackEndpoints.cs

```diff
@@ -0,0 +1,206 @@
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Routing;
+using StackFrame = Microsoft.ApplicationInsights.DataContracts.StackFrame;
+
+namespace PlatformPlatform.SharedKernel.ApiCore.Endpoints;
+
+public static class TrackEndpoints
+{
+    public static void MapTrackEndpoints(this IEndpointRouteBuilder routes)
+    {
+        routes.MapPost("/api/track", Track);
+    }
+
+    private static TrackResponseSuccessDto Track(
+        List<TrackRequestDto> trackRequests,
+        TelemetryClient telemetryClient,
+        ILogger<string> logger
+    )
+    {
+        foreach (var trackRequestDto in trackRequests)
+        {
+            switch (trackRequestDto.Data.BaseType)
+            {
+                case "PageviewData":
+                {
+                    var telemetry = new PageViewTelemetry
+                    {
+                        Name = trackRequestDto.Data.BaseData.Name,
+                        Url = new Uri(trackRequestDto.Data.BaseData.Url),
+                        Duration = trackRequestDto.Data.BaseData.Duration,
+                        Timestamp = trackRequestDto.Time,
+                        Id = trackRequestDto.Data.BaseData.Id
+                    };
+
+                    CopyContextTags(telemetry.Context, trackRequestDto);
+                    CopyDictionary(trackRequestDto.Data.BaseData.Properties, telemetry.Properties);
+                    CopyDictionary(trackRequestDto.Data.BaseData.Measurements, telemetry.Metrics);
+
+                    telemetryClient.TrackPageView(telemetry);
+                    break;
+                }
+                case "PageviewPerformanceData":
+                {
+                    var telemetry = new PageViewPerformanceTelemetry
+                    {
+                        Name = trackRequestDto.Data.BaseData.Name,
+                        Url = new Uri(trackRequestDto.Data.BaseData.Url),
+                        Duration = trackRequestDto.Data.BaseData.Duration,
+                        Timestamp = trackRequestDto.Time,
+                        Id = trackRequestDto.Data.BaseData.Id,
+                        PerfTotal = trackRequestDto.Data.BaseData.PerfTotal,
+                        NetworkConnect = trackRequestDto.Data.BaseData.NetworkConnect,
+                        SentRequest = trackRequestDto.Data.BaseData.SentRequest,
+                        ReceivedResponse = trackRequestDto.Data.BaseData.ReceivedResponse,
+                        DomProcessing = trackRequestDto.Data.BaseData.DomProcessing
```

Comment:
I noticed a lot of extra properties was added. I have not reviewed all the different Telemetry items, but are we sure that the data we track with this end point has the same data as if we track it using the SDK?